File: tasks-backend/src/domain/accounts/repositories.py
import uuid
from typing import Optional
from src.domain import BaseRepository
from src.domain.accounts import Account


class AccountRepository(BaseRepository):

    def get_by_id(self, unit_of_work, account_id: uuid) -> Optional[Account]:
        raise NotImplemented("AccountRepository#get_by_id is not implemented.")

    def get_by_email(self, unit_of_work, email: str) -> Optional[Account]:
        raise NotImplemented("AccountRepository#get_by_email is not implemented")

    def insert(self, unit_of_work, account: Account) -> uuid:
        raise NotImplemented("AccountRepository#insert is not implemented.")

    def update(self, unit_of_work, account: Account) -> None:
        raise NotImplemented("AccountRepository#update is not implemented.")

    def delete(self, unit_of_work, account_id: uuid) -> None:
        raise NotImplemented("AccountRepository#delete is not implemented.")
    

File: tasks-desktop/src/infrastructure/modules/accounts/repositories.js
const { Account } = require("../../../domain/modules/accounts");
const { Tables, Fields } = require("../../persistence/tables");

const save = async (unitOfWork, account) => {
    const queryManager = unitOfWork.getQueryManager();
    await queryManager.run(
        `INSERT INTO ${Tables.Accounts}(${Fields.Accounts.Id}, ${Fields.Accounts.Email}, ${Fields.Accounts.Language})`
        + `VALUES (?, ?, ?)`
        , [account.id, account.email, account.language]);
}

const getByEmail = async (unitOfWork, email) => {
    const queryManager = unitOfWork.getQueryManager();
    
    const queryResult = await queryManager.get(
        `SELECT * `
        + `FROM ${Tables.Accounts} `
        + `WHERE ${Fields.Accounts.Email} = ?`
        , [email]);
    return queryResult != undefined 
            ? new Account(queryResult.id, queryResult.email, queryResult.language) 
            : undefined;
}

const get = async (unitOfWork, accountId) => {
    const queryManager = unitOfWork.getQueryManager();
    const queryResult = await queryManager.get(
        `SELECT * `
        + `FROM ${Tables.Accounts} `
        + `WHERE ${Fields.Accounts.Id} = ?`
        , [accountId]);

    return queryResult != undefined 
            ? new Account(queryResult.id, queryResult.email, queryResult.language) 
            : undefined;
}

const setLanguage = async (unitOfWork, accountId, language) => {
    const queryManager = unitOfWork.getQueryManager();

    await queryManager.run(
        `UPDATE ${Tables.Accounts} SET ${Fields.Accounts.Language}=? `
        + `WHERE ${Fields.Accounts.Id}=? `
        , [language, accountId]
    );
}

module.exports.AccountRepository = {
    save
    , getByEmail
    , get
    , setLanguage
};